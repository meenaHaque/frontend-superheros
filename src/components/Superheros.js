import React, { useEffect, useState } from "react";
//import superheros from "../Superheros";

const listStyles = {
	color: "green",
	fontSize: "16px",
	padding: "10px 2px",
};

const Superheros = () => {
	const [superheros, setSuperheros] = useState([]);

	useEffect(() => {
		fetch("http://localhost:3001/api/v1/superheros")
			.then((response) => response.json())
			.then((foo) => {
				setSuperheros(foo);
			});
	});

	return (
		<ol>
			{superheros.map((hero) => {
				return (
					<li key={hero.id} style={listStyles}>
						{hero.name} - {hero.description}
					</li>
				);
			})}
		</ol>
	);
};

export default Superheros;
