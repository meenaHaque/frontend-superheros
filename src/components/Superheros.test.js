import React from 'react';
import renderer from 'react-test-renderer';
import { render } from '@testing-library/react';
import Superheros from './Superheros';

describe('Superheroes component', () => {
  test('matches the superheroes snapshot', () => {
    const component = renderer.create(<Superheros />);
    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });

  test('Batwoman is in the list', () => {
    const { getByText } = render(<Superheros />);
    const batwoman = getByText(/batwoman/i);
    expect(batwoman).toBeInTheDocument();
  });
});
