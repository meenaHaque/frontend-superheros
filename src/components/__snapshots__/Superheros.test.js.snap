// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Superheroes component matches the superheroes snapshot 1`] = `
<ol>
  <li
    style={
      Object {
        "color": "green",
        "fontSize": "16px",
        "padding": "10px 2px",
      }
    }
  >
    ANT-MAN
     - 
    Ability to shrink to sub-microscopic size
  </li>
  <li
    style={
      Object {
        "color": "green",
        "fontSize": "16px",
        "padding": "10px 2px",
      }
    }
  >
    BATWOMAN
     - 
    Highly skilled martial artis
  </li>
  <li
    style={
      Object {
        "color": "green",
        "fontSize": "16px",
        "padding": "10px 2px",
      }
    }
  >
    CATWOMAN
     - 
    Skilled gymnast
  </li>
  <li
    style={
      Object {
        "color": "green",
        "fontSize": "16px",
        "padding": "10px 2px",
      }
    }
  >
    IRON MAN
     - 
    Proficient scientist and engineer
  </li>
  <li
    style={
      Object {
        "color": "green",
        "fontSize": "16px",
        "padding": "10px 2px",
      }
    }
  >
    BLACK PANTHER
     - 
    Superhuman strength, speed, reflexes
  </li>
  <li
    style={
      Object {
        "color": "green",
        "fontSize": "16px",
        "padding": "10px 2px",
      }
    }
  >
    CAPTAIN AMERICA
     - 
    Agility, strength, speed, endurance
  </li>
</ol>
`;
